core/src/main/java/org/keycloak/util/EnumWithUnchangableIndex.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.util;
+
+/**
+ * Interface for marking {@link Enum}s to have an index.
+ * The index might be used for persisting so that particular item index 
+ * cannot be changed.
+ */
+public interface EnumWithUnchangableIndex {
+    public Integer getUnchangebleIndex();    
```

Comment:
No, the `Integer` came from assumption that we'd store `Enum`s numeric representations on entity level. In that case we'd want to have possibility to store `null`s due to tree store and store hierarchy.

But now, since it has been decided we let each store handle it separately it seems it might make sense to have `int` there. I'll update it.